File: Random Password Generator/password.py
import tkinter as tk
from tkinter import messagebox
import random
import string
import pyperclip

def generate_password():
    letter_limit = int(entry_letter_limit.get())
    lowercase_enabled = chk_var_lower.get()
    uppercase_enabled = chk_var_upper.get()
    special_characters_enabled = chk_var_special.get()

    character_set = ""
    if lowercase_enabled:
        character_set += string.ascii_lowercase
    if uppercase_enabled:
        character_set += string.ascii_uppercase
    if special_characters_enabled:
        character_set += "!@#$%^&*()-_=+{[]}"

    if not character_set:
        messagebox.showerror("Error", "You must select at least one type of character.")
        return

    password = ''.join(random.choice(character_set) for _ in range(letter_limit))
    lbl_generated_password.config(text="The password is: " + password)

    btn_copy_password.config(state=tk.NORMAL)

def copy_password():
    password = lbl_generated_password.cget("text")[16:]  # Extract password from label text
    pyperclip.copy(password)
    messagebox.showinfo("Password Copied", "Password copied to clipboard!")

# Create the tkinter window
window = tk.Tk()
window.title("Password Generator")

# Letter Limit Entry
lbl_letter_limit = tk.Label(window, text="Enter the letter limit: ")
lbl_letter_limit.pack()
entry_letter_limit = tk.Entry(window)
entry_letter_limit.pack()

# Checkboxes for character types
chk_var_lower = tk.IntVar()
chk_var_upper = tk.IntVar()
chk_var_special = tk.IntVar()

chk_lower = tk.Checkbutton(window, text="Lowercase letters", variable=chk_var_lower)
chk_upper = tk.Checkbutton(window, text="Uppercase letters", variable=chk_var_upper)
chk_special = tk.Checkbutton(window, text="Special characters", variable=chk_var_special)

chk_lower.pack()
chk_upper.pack()
chk_special.pack()

# Generate Password Button
btn_generate = tk.Button(window, text="Generate Password", command=generate_password)
btn_generate.pack()

# Generated Password Label
lbl_generated_password = tk.Label(window, text="")
lbl_generated_password.pack()

# Copy Password Button
btn_copy_password = tk.Button(window, text="Copy Password", state=tk.DISABLED, command=copy_password)
btn_copy_password.pack()

window.mainloop()

File: Random Password Generator/password.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() 
{
    int letter_limit;
    printf("Enter the letter limit: ");
    scanf("%d", &letter_limit);

    char password[letter_limit + 1];

    srand(time(NULL));

    char lowercase_characters[] = "abcdefghijklmnopqrstuvwxyz";
    char uppercase_characters[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    char special_characters[] = "!@#$%^&*()-_=+{[]}";

    int lowercase_enabled = 0;
    int uppercase_enabled = 0;
    int special_characters_enabled = 0;

    printf("Do you want lowercase letters in the password? (y/n): ");
    char input;
    scanf(" %c", &input);

    if (input == 'y' || input == 'Y') 
    {
        lowercase_enabled = 1;
    }

    printf("Do you want uppercase letters in the password? (y/n): ");
    scanf(" %c", &input);

    if (input == 'y' || input == 'Y') 
    {
        uppercase_enabled = 1;
    }

    printf("Do you want special characters in the password? (y/n): ");
    scanf(" %c", &input);

    if (input == 'y' || input == 'Y') 
    {
        special_characters_enabled = 1;
    }

    int character_set_size = 0;
    if (lowercase_enabled) character_set_size += sizeof(lowercase_characters) - 1;
    if (uppercase_enabled) character_set_size += sizeof(uppercase_characters) - 1;
    if (special_characters_enabled) character_set_size += sizeof(special_characters) - 1;

    if (character_set_size == 0) 
    {
        printf("You must select at least one type of character.\n");
        return 1;
    }

    for (int i = 0; i < letter_limit; i++) 
    {
        int random_character;
        int set_choice = rand() % character_set_size;

        if (lowercase_enabled && set_choice < sizeof(lowercase_characters) - 1) 
        {
            random_character = lowercase_characters[rand() % (sizeof(lowercase_characters) - 1)];
        } 
        else if (uppercase_enabled && set_choice < (sizeof(lowercase_characters) - 1) + (sizeof(uppercase_characters) - 1)) {
            random_character = uppercase_characters[rand() % (sizeof(uppercase_characters) - 1)];
        } 
        else if (special_characters_enabled) 
        {
            random_character = special_characters[rand() % (sizeof(special_characters) - 1)];
        } 
        else 
        {
            random_character = lowercase_characters[rand() % (sizeof(lowercase_characters) - 1)];
        }

        password[i] = random_character;
    }

    password[letter_limit] = '\0';

    printf("The password is: %s\n", password);

    printf("------------Press Ok To Exit------------\n");
    fflush(stdout);
    getchar();    

    char n12;
    scanf("%c",&n12);

    return 0;
}